contracts/HydraChain/modules/Inspector/Inspector.sol

```diff
@@ -38,14 +43,46 @@ abstract contract Inspector is IInspector, ValidatorManager {
 
     // _______________ External functions _______________
 
+    /**
+     * @inheritdoc IInspector
+     */
+    function initiateBan(address validator) external {
+        if (bansInitiated[validator] != 0) {
+            revert BanAlreadyInitiated();
+        }
+
+        if (!isSubjectToInitiateBan(validator)) {
+            revert NoInitiateBanSubject();
+        }
+
+        bansInitiated[validator] = block.timestamp;
+        hydraStakingContract.temporaryRemove(validator);
+    }
+
+    /**
+     * @inheritdoc IInspector
+     */
+    function terminateBanProcedure() external {
```

Comment:
What if the validator calls terminateBanProcedure but still does not sign blocks and is inactive? We can initiateBan again, but the timer will reset.

Reply:
It is fine to reset the timer because the staker made an action by terminating the ban. If this behavior is repeated more than once, we could propose a governance ban because of malicious behavior. However the idea of this automatic ban mechanism is clearing offline validators, not protecting the chain from malicious behavior.